src/plugin/dl/dlwrappers.cpp

```diff
@@ -136,15 +136,22 @@ dlopen_try_paths(const char *filename, int flag, string path)
 extern "C"
 void *dlopen(const char *filename, int flag)
 {
+  // TODO(kapil): Replace with scoped lock once we have it.
   bool lockAcquired = dmtcp_libdlLockLock();
   string rpath;
   string runpath;
   void *ret = NULL;
 
   getRpathRunPath(__builtin_return_address(0), &rpath, &runpath);
 
-  // 1. Attempt dlopen using RPATH.
-  if (!rpath.empty()) {
+  if (filename == NULL) {
+    ret = _real_dlopen(filename, flag);
+  }
```

Comment:
Sorry to be picky, but isn't the above code redundant?  If `filename == NULL`, then the code below will already do:
> `ret = _real_dlopen(filename, flag);`